app/Models/User.php

```diff
@@ -213,27 +213,28 @@ public function resumeURL($method = 'get')
         if (App::environment() == 'testing') {
             return "http://s3-mock-resumes/{$this->id}.pdf";
         }
-        $s3 = AWS::createClient('s3');
-        switch ($method) {
-            case 'get':
-                $cmd = $s3->getCommand('getObject', [
-                    'Bucket' => getenv('S3_BUCKET'),
-                    'Key'    => getenv('S3_PREFIX').'/resumes/'.$this->id.'.pdf',
-                    'ResponseContentType' => 'application/pdf',
-                ]);
-                break;
-            case 'put':
-                $cmd = $s3->getCommand('PutObject', [
-                    'Bucket' => getenv('S3_BUCKET'),
-                    'Key'    => getenv('S3_PREFIX').'/resumes/'.$this->id.'.pdf',
-                ]);
-                break;
+
+        $expiry = '+7 days';
+
+        if ($method == 'get') {
+            return Storage::cloud()->temporaryUrl($this->getResumeFilePath(), now()->modify($expiry), ['ResponseContentType' => 'application/pdf']);
         }
-        $request = $s3->createPresignedRequest($cmd, '+7 days');
+
+        $client = Storage::cloud()->getDriver()->getAdapter()->getClient();
+        $command = $client->getCommand('PutObject', [
+            'Bucket' => getenv('AWS_BUCKET'),
+            'Key'    => $this->getResumeFilePath(),
+        ]);
+        $request = $client->createPresignedRequest($command, $expiry);
 
         return (string) $request->getUri();
```

Comment:
can get rid of the string cast as storage layer already does that
https://github.com/illuminate/filesystem/blob/master/FilesystemAdapter.php#L534

Reply:
We would still need this, since PutObject doesn't use the getAwsTemporaryUrl method.